tests/DotNetBoilerplate.Tests.Integration/BookStoreEndpointsTests.cs

```diff
@@ -0,0 +1,248 @@
+
+        var bookStores = await dbContext.BookStores.ToListAsync();
+        bookStores.Count.ShouldBe(1);
+
+        var bookStore = await dbContext.BookStores.FirstOrDefaultAsync();
+        bookStore.ShouldNotBeNull();
+        bookStore!.Name.ShouldBe(existingBookStore.Name);
+    }
+
+    public async Task GivenBookStoreExists_UpdateBookStore_ShouldSucceed()
+    {
+        //Arrange
+        var createRequest = new CreateBookStoreEndpoint.Request
+        {
+            Name = "BookStore",
+            Description = "Description"
+        };
+        var createResponse = await testsFixture.Client.PostAsJsonAsync("book-stores", createRequest);
+        createResponse.StatusCode.ShouldBe(HttpStatusCode.OK);
+
+        using var serviceScope = testsFixture.ServiceProvider.CreateScope();
+        var dbContext = serviceScope.ServiceProvider.GetRequiredService<DotNetBoilerplateReadDbContext>();
+
+        var bookStore = await dbContext.BookStores.FirstOrDefaultAsync();
+        bookStore.ShouldNotBeNull();
+        bookStore!.Name.ShouldBe(createRequest.Name);
+
+
+        var updateRequest = new UpdateBookStoreEndpoint.Request
+        {
+            Name = "Updated book store",
+            Description = "Updated description"
+        };
+
+        //Act
+        var updateResponse = await testsFixture.Client.PutAsJsonAsync($"book-stores/{bookStore.Id}", updateRequest);
+
+         //Assert
+        updateResponse.StatusCode.ShouldBe(HttpStatusCode.OK);
+
+        var updatedBookStore = await dbContext.BookStores.FirstOrDefaultAsync(x=>x.Id==bookStore.Id);
+
+        updatedBookStore.ShouldNotBeNull();
+        updatedBookStore!.Name.ShouldBe(updateRequest.Name);
+    }
+
+    [Fact]
+    public async Task GivenBookStoreDoesNotExists_UpdateBookStore_ShouldFail()
+    {
+        //Arrange
+        var updateRequest = new UpdateBookStoreEndpoint.Request
+        {
+            Name = "Updated book store",
+            Description = "Updated description"
+        };
+
+        var randomId = Guid.NewGuid();
+        //Act
+        var updateResponse = await testsFixture.Client.PutAsJsonAsync($"book-stores/{randomId}", updateRequest);
+
+         //Assert
+        updateResponse.StatusCode.ShouldBe(HttpStatusCode.BadRequest);
+    }
+
+    [Fact]
+    public async Task GivenBookStoresExists_BrowseBookStores_ShouldReturnAll()
+    {
+        //Arrange
+        var firstBookStore = new CreateBookStoreEndpoint.Request
+        {
+            Name = "BookStore",
+            Description = "Description"
+        };
+
+        var secondBookStore = new CreateBookStoreEndpoint.Request
+        {
+            Name = "BookStore2",
+            Description = "Description2"
+        };
+
+        //Act
+        var firstResponse = await testsFixture.Client.PostAsJsonAsync("book-stores", firstBookStore);
+
+        testsFixture.Client.DefaultRequestHeaders.Remove("Authorization");
```

Comment:
możemy stworzyć sobie do tego metodę np to extension method bo możliwe że użyjemy jeszcze kilka razy